### DISCUSSION TITLE
Restrict debugging to certain packet types is not supported?

### DISCUSSION THREAD

**Discussion by polad** [sidorares/node-mysql2#1759] [Q&A]
It seems like the ability to restrict debugging to certain packet types is not supported as in `mysqljs/mysql`.
From the docs: [mysqljs/mysql - Debugging and reporting problems](https://www.npmjs.com/package/mysql#debugging-and-reporting-problems)

It's possible in `mysqljs/mysql` to pass an array of packet types (i.e. packet constructor names) using the `debug` option:
```js
mysql.createConnection({debug: ['ComQueryPacket', 'RowDataPacket']});
```
This was implemented in `mysqljs/mysql` a while back. Here is the PR from 2013: mysqljs/mysql#413

I looked through the code under `mysql2/lib` and all I could find was only checks for a boolean flag like:
```js
if (this.config.debug) {
   console.log(...
```
I was hoping to see in debug output all SQL queries that `mysql2` sends to the server but setting `debug: true` prints too much noise like client/server handshakes, raw data packets, etc..

Any suggestions how I can print only the SQL queries in debug output?

Thanks!

**Comment by tafadzwagonera** [comment]
@polad Did you find a way around this issue?

**Reply by polad** [reply]
@tafadzwagonera It's not really an issue. I ended up using a custom wrapper function.

Assuming you have some function that instantiates a MySQL pool or connection like so:
```javascript
const dbConnect = () => { /* returns MySQL Pool instance or Connection */ }
```
you could do the following in a wrapper function:
```javascript
const query = (sql, params) => {
  console.debug(">> QUERY:", sql, params);
  return dbConnect.query(sql, params);
};
```
Then you use the above `query()` function anywhere in your code it would print your SQL and query parameters. I used `console.log` here for brevity but you can use a custom logger if you want this to be configurable based on your environments like local dev, production, etc. Just do it in your custom `query()` function.

**Reply by tafadzwagonera** [reply]
Thank you. I'm using `mysql2` through `typeorm` and `@nestjs/typeorm` wrappers which hide the native `query` capability with domain abstraction. I was hoping the `debug` configuration would provide capability to fine tune the logs but I'm struggling to get lean logs. Perhaps my assumption about how `debug` is supposed to work is incorrect. I don't know anymore.

**Reply by polad** [reply]
I raised this question initially because I wanted to see what SQL queries/parameters are executed by my code. Since the `debug` config of the `mysql2` was too noisy for my needs I opted out for a custom wrapper function.

In your case looks like TypeORM supports logging with various options: https://typeorm.io/logging

**Reply by tafadzwagonera** [reply]
I've configured TypeORM logging in conjuction with `mysql2` driver `debug`. It's working as expected. I thought configuring `mysql2` with the following configuration `debug: ['ComQueryPacket', 'ResultSetHeaderPacket', 'RowDataPacket']` would help provide further visibility into queries and result sets but I'm getting a tonne of information which drowns out relevant information and makes debugging harder than it should be.
